### ISSUE TITLE
annotation querying

### ISSUE THREAD

**Issue by thehabes** [CenterForDigitalHumanities/deer#7]
`findByTargetId()` will not find annotations that target a resource like

```javascript
{
  "type": "Annotation",
  "label":"Gender",
  "schemaKey":"gender",
  "body": {
    "value": "Male",
    "format" : "text/html"
  },
  "target": {
    "@id": "http://devstore.rerum.io/v1/id/5cbdd3ace4b0511ce825256e"
  }
}
```
because findByTargetId builds it's query object to look for annotations like 

`"target":"http://devstore.rerum.io/v1/id/5cbdd3ace4b0511ce825256e"`

We need to let findByTargetId query for objects that target a resource in "any" web annotation compliant way.

https://www.w3.org/TR/annotation-model/#bodies-and-targets

**Comment by thehabes** [comment]
```javascript
//Give me all objects that target a resource's id in web annotation compliant ways
{ 
   $or: [
      { "target" : "http://devstore.rerum.io/v1/id/5cbdd3ace4b0511ce825256e" },
      { "target.id" : "http://devstore.rerum.io/v1/id/5cbdd3ace4b0511ce825256e" },
      { "target.@id" : "http://devstore.rerum.io/v1/id/5cbdd3ace4b0511ce825256e" },
      { "target.source" : "http://devstore.rerum.io/v1/id/5cbdd3ace4b0511ce825256e" }
   ]
}
```

**Comment by thehabes** [comment]
```javascript
/**
     * Execute query for any annotations in RERUM which target the
     * id passed in. Promise resolves to an array of annotations.
     * @param {String} id URI for the targeted entity
     */
    findByTargetId: async function (id) {
        let everything = Object.keys(localStorage).map(k => JSON.parse(localStorage.getItem(k)))
        let obj = {"$or":[]}
        obj["$or"].push({"target":id})
        obj["$or"].push({"target.id":id})
        obj["$or"].push({"target.@id":id})
        //obj["$or"].push({"target.source": id})
        let matches = await fetch(DEER.URLS.QUERY, {
            method: "POST",
            body: JSON.stringify(obj),
            headers: {
                "Content-Type": "application/json"
            }
        })
            .then(response => response.json())
            .catch((err) => console.log(err))
        let local_matches = everything.filter(o => o.target === id)
        matches = local_matches.concat(matches)
        return matches
    }
```

**Comment by thehabes** [comment]
This has been resolved through #8, awaiting review and acceptance from @cubap
